deepcell/model_zoo/fpn.py

```diff
@@ -461,104 +488,52 @@ def __create_semantic_head(pyramid_dict,
     pyramid_names.reverse()
     pyramid_features.reverse()
 
-    semantic_features = []
-    semantic_names = []
+    # Previous method of building feature pyramids
+    # semantic_features, semantic_names = [], []
+    # for N, P in zip(pyramid_names, pyramid_features):
+    #     # Get level and determine how much to upsample
+    #     level = int(re.findall(r'\d+', N)[0])
+    #
+    #     n_upsample = level - target_level
+    #     target = semantic_features[-1] if len(semantic_features) > 0 else None
+    #
+    #     # Use semantic upsample to get semantic map
+    #     semantic_features.append(semantic_upsample(
+    #         P, n_upsample, n_filters=n_filters, target=target, ndim=ndim,
+    #         upsample_type=upsample_type, interpolation=interpolation,
+    #         semantic_id=semantic_id))
+    #     semantic_names.append('Q{}'.format(level))
+
+    # Add all the semantic features
+    # semantic_sum = semantic_features[0]
+    # for semantic_feature in semantic_features[1:]:
+    #     semantic_sum = Add()([semantic_sum, semantic_feature])
+
+    semantic_sum = pyramid_features[-1]
 
-    for N, P in zip(pyramid_names, pyramid_features):
-        # Get level and determine how much to upsample
-        level = int(re.findall(r'\d+', N)[0])
+    # Final upsampling
+    min_level = int(re.findall(r'\d+', pyramid_names[-1])[0])
+    # n_upsample = min_level - target_level
+    n_upsample = min_level
```

Comment:
I was unsure about this.

Reply:
Both `RetinaNet(panoptic=True)` and `PanopticNet` create semantic heads with `target_level` calculated as:

```python
pyramid_dict = create_pyramid_features(...)
semantic_levels = [int(re.findall(r'\d+', k)[0]) for k in pyramid_dict]
target_level = min(semantic_levels)
```

With the current code above, we ignore `target_level` and calculates `n_upsample` for the final semantic prediction without `target_sample`.

```python
min_level = int(re.findall(r'\d+', pyramid_names[-1])[0])
# n_upsample = min_level - target_level
n_upsample = min_level
```

I think that `target_level` will always be equal to `min_level`. Should the final n_upsample usually be `0`? If not, can we just shortcut to `n_upsample = target_level`?